rom/dev/src/flow/cold_reset/fw_processor.rs

```diff
@@ -129,30 +127,40 @@ impl FirmwareProcessor {
         cprint!("[afmc] Waiting for Image ");
         loop {
             if let Some(txn) = mbox.peek_recv() {
-                if txn.cmd() != Self::MBOX_DOWNLOAD_FIRMWARE_CMD_ID {
-                    cprintln!("Invalid command 0x{:08x} received", txn.cmd());
-                    txn.start_txn().complete(false)?;
-                    return Err(CaliptraError::FW_PROC_MAILBOX_INVALID_COMMAND);
-                }
-
-                // Re-borrow mailbox to work around https://github.com/rust-lang/rust/issues/54663
-                let txn = mbox
-                    .peek_recv()
-                    .ok_or(CaliptraError::FW_PROC_MAILBOX_STATE_INCONSISTENT)?;
-
-                // This is a download-firmware command; don't drop this, as the
-                // transaction will be completed by either handle_fatal_error() (on
-                // failure) or by a manual complete call upon success.
-                let txn = ManuallyDrop::new(txn.start_txn());
-                if txn.dlen() == 0 || txn.dlen() > IMAGE_BYTE_SIZE as u32 {
-                    cprintln!("Invalid Image of size {} bytes" txn.dlen());
-                    return Err(CaliptraError::FW_PROC_INVALID_IMAGE_SIZE);
+                match txn.cmd() {
+                    MBOX_CMD_ID_DOWNLOAD_FIRMWARE => {
+                        // Re-borrow mailbox to work around
+                        // https://github.com/rust-lang/rust/issues/54663
+                        let txn = mbox
+                            .peek_recv()
+                            .ok_or(CaliptraError::FW_PROC_MAILBOX_STATE_INCONSISTENT)?;
+
+                        // This is a download-firmware command; don't drop this, as
+                        // the transaction will be completed by either
+                        // handle_fatal_error() (on failure) or by a manual complete
+                        // call upon success.
+                        let txn = ManuallyDrop::new(txn.start_txn());
+                        if txn.dlen() == 0 || txn.dlen() > IMAGE_BYTE_SIZE as u32 {
+                            cprintln!("Invalid Image of size {} bytes" txn.dlen());
+                            return Err(CaliptraError::FW_PROC_INVALID_IMAGE_SIZE);
+                        }
+
+                        cprintln!("");
+                        cprintln!("[afmc] Received Image of size {} bytes" txn.dlen());
+                        report_boot_status(FwProcessorDownloadImageComplete.into());
+                        return Ok(txn);
+                    }
+                    MBOX_CMD_ID_RESERVED_O | MBOX_CMD_ID_RESERVED_1 | MBOX_CMD_ID_RESERVED_2 => {
+                        cprintln!("Reserved command 0x{:08x} received", txn.cmd());
+                        txn.start_txn().complete(false)?;
```

Comment:
I assume we want to add a call to report_fw_error_non_fatal(CaliptraError::ROM_UPDATE_RESET_FLOW_INVALID_FIRMWARE_COMMAND.into()) here?